mining.py

```diff
@@ -0,0 +1,175 @@
+import pickle
+import wikipedia
+import math
+input_file = open('britentries.pickle', 'rb')
+encyclopedia = pickle.load(input_file)
+
+
+
+
+
+#finds index of next char
+def find_next_char(i, string):
+    while True:
+        if string[i] == ' ' or string[i] == '\n':
+            i += 1
+        else:
+            return i
+
+#finds name of entry, must start with chars
+def find_entry(i, string):
+    entry = ''
+    while True: #isolates entry titles
+        if string[i] == '\n' or string[i:i+2] == '  ' or string[i:i+2] == '. ':
+            break
+        else:
+            entry += string[i]
+            i += 1
+    return entry
+#i_slice = find_next_char(i_slice) #determines index of first entry in table of contents
+
+
+
+ #adds all entries to contents list
+def fill_contents(i, string):
+    contents = []
+    while True:
+        entry = find_entry(i, string)
+        i += len(entry)
+        if len(contents) > 1 and contents[0] == entry:
+            break
+        contents.append(entry)
+        i = find_next_char(i, string)
+    #contents = contents.sort()
+    return contents
+
+#checks if entry is name, if it is name, returns correct format, otherwise returns orgininal entry
+def check_name(entry):
+    comma = entry.find(',')
+    if comma > - 1:
+        entry = entry[comma+2:] + entry[:comma]
+    return entry
+
+#creates list of entries of texts
+def fill_text_list(i, string, contents):
+    texts = []
+    for entry in contents:
+        original = entry
+        if entry.find('(') > -1:
+            entry = entry[:entry.index('(') - 1]
+        if len(entry) > 12:
```

Comment:
Try to stay away from magic numbers (in this case, 12). Store it in a variable first, so we know why you are using that number